### PR TITLE
Paginated Directory Delete

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs [hunk 1]
```diff
@@ -1875,31 +1875,40 @@ internal virtual void SetNameFieldsIfNull()
                 try
                 {
                     scope.Start();
-                    ResponseWithHeaders<PathDeleteHeaders> response;
+                    ResponseWithHeaders<PathDeleteHeaders> response = null;
+                    bool paginated = _clientConfiguration.ClientOptions.Version >= DataLakeClientOptions.ServiceVersion.V2023_05_03;
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs:1879]: Waiting to hear back from the service if this is really the min version for this feature.

File: sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs [hunk 2]
```diff
@@ -1019,6 +1019,28 @@ public async Task DeleteAsync_ConditionsFail()
             }
         }
 
+        [RecordedTest]
+        //[LiveOnly]
+        [ServiceVersion(Min = DataLakeClientOptions.ServiceVersion.V2023_08_03)]
+        public async Task DeleteAsync_Paginated()
```
[the remark this answers is not in the document]
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs:1025]: This test needs to use OAuth.  Waiting on this feature to be deployed to canary.

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs [hunk 3]
```diff
@@ -1875,31 +1875,47 @@ internal virtual void SetNameFieldsIfNull()
...
+                                ifModifiedSince: conditions?.IfModifiedSince,
+                                ifUnmodifiedSince: conditions?.IfUnmodifiedSince,
+                                paginated: paginated,
+                                cancellationToken: cancellationToken);
+                        }
                     }
+                    while (!string.IsNullOrEmpty(response?.Headers?.Continuation));
 
                     return response.GetRawResponse();
                 }
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs:1921]: Consider including continuation token if there is an exception

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs:1921]: Also consider allowing customer to pass in continuation token in this API.

(Talk to service team and see if this scenario is something we care about)

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs:1921]: We also may want to consider progress tracking (possibly in the future)

File: sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs [hunk 4]
```diff
@@ -1875,31 +1875,47 @@ internal virtual void SetNameFieldsIfNull()
                 try
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/src/DataLakePathClient.cs:1875]: Consider taking an optional IProgress<T> as a parameter.

File: sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeClientTestFixtureAttribute.cs [hunk 5]
```diff
@@ -32,7 +32,7 @@ public DataLakeClientTestFixtureAttribute()
                 StorageVersionExtensions.MaxVersion)
         {
             RecordingServiceVersion = StorageVersionExtensions.MaxVersion;
-            LiveServiceVersions = new object[] { StorageVersionExtensions.LatestVersion };
+            LiveServiceVersions = new object[] { StorageVersionExtensions.MaxVersion };
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.DataLake/tests/DataLakeClientTestFixtureAttribute.cs:35]: TODO revert this.

**Review by amnguye — APPROVED** [review]

**Pull request merged**